src/RequestAuthenticator.php

```diff
@@ -71,7 +71,7 @@ public function authenticate(RequestInterface $request)
         $compareSignature = $compareAuthHeader->getSignature();
 
 
-        if ($signature !== $compareSignature) {
+        if (!hash_equals($signature, $compareSignature)) {
```

Comment:
Please double check the parameter order here, and at the other use of this function, I think it's incorrect now, but I might be wrong.
As the PHP doc says: "It is important to provide the user-supplied string as the second parameter, rather than the first."